src/pages/HomePage/HomePage.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router';
+import PageLayout from '../../PageLayout/PageLayout';
+import FilterableList from '../../components/FilterableList/FilterableList';
+import getCitiesData from '../../services/getCitiesData';
+import { minValueLength } from '../../common/constants';
+
+function HomePage() {
+  const [cities, setCities] = useState([]);
+  const [inputValue, setInputValue] = useState('');
+  const history = useHistory();
+
+  const getAllCitiesByName = async event => {
+    setInputValue(event.target.value.toLowerCase());
+
+    if (inputValue.length >= minValueLength) {
+      try {
+        let getDataCities = await getCitiesData(inputValue);
+        setCities(getDataCities);
+      } catch (error) {
+        console.log(error);
+        history.push('/error');
+      }
+    } else {
+      setCities([]);
```

Comment:
should you check the fact that cities are already set to [] to reduce number of updates ?

Reply:
If the user clear the input value I want cities to be set it to [] and FilterableList component to be re-rendered with empty array not to be shown with previous state.